src/Mapper.php

```diff
@@ -1,11 +1,156 @@
 <?php
 
-namespace Tnapf\JsonMapper\Mapper;
+namespace Tnapf\JsonMapper;
+
+use ReflectionAttribute;
+use ReflectionClass;
+use ReflectionProperty;
+use Tnapf\JsonMapper\Attributes\AnyArray;
+use Tnapf\JsonMapper\Attributes\AnyType;
+use Tnapf\JsonMapper\Attributes\CaseConversionInterface;
+use Tnapf\JsonMapper\Attributes\BaseType;
+use Tnapf\JsonMapper\Attributes\BoolType;
+use Tnapf\JsonMapper\Attributes\FloatType;
+use Tnapf\JsonMapper\Attributes\IntType;
+use Tnapf\JsonMapper\Attributes\ObjectArrayType;
+use Tnapf\JsonMapper\Attributes\ObjectType;
+use Tnapf\JsonMapper\Attributes\StringType;
 
 class Mapper implements MapperInterface
 {
+    protected ?CaseConversionInterface $caseConversion = null;
+    protected object $instance;
+    protected ReflectionClass $reflection;
+
+    /**
+     * @var BaseType[]
+     */
+    protected array $attributes;
+
+    protected string $class;
+    protected array $data;
+
     public function map(string $class, array $data): object
     {
-        $instance = new $class();
+        $instance = (new self());
+        $instance->class = $class;
+        $instance->data = $data;
+        $instance->reflection = new ReflectionClass($class);
```

Comment:
The initialization should be done in constructor in this case (lines 36-38).

Reply:
Yes it should, however that is not possible sadly which is why I'm initializing it like this. But I don't see any real issue besides it being abnormal.